### ISSUE TITLE
Unable to make (or run) a new project (windows 10)

### ISSUE THREAD

**Issue by guessWho8** [NativeScript/nativescript-cli#2921]
Hello,
many windows users are having same problem as me. I'm not able to create or run new project. Detailed description is available [here](https://discourse.nativescript.org/t/unable-to-use-nativescript-cli/1748).

### Which platform(s) does your issue occur on?
Both

### Please provide the following version numbers that your issue occurs with:
- CLI: 3.0.3
- Cross-platform modules: I don't have any project folder
- Runtime(s): I don't have any project folder
- Plugin(s): I don't have any project folder

### Please tell us how to recreate the issue in as much detail as possible.
You can find everything in discourse link. I tried to debug the problem using log trace option and it looks like there's something wrong with project path. It's always looking for (and installing) npm modules in current directory instead of the project one.

**Comment by rosen-vladimirov** [comment]
Hey @guessWho8 ,
Thank you for the detailed logs and description of the issue. I've investigated them and I've tried to reproduce the issue, but with no luck.
May I ask you to try the following and provide the requested information below:
1. Create a new directory on a different location (not under your home dir, in case you have a D: drive it will be best to try there). In the new directory execute `tns create myApp --ng --log trace` and send us the output in case it failse.

2. Do you have a `package.json` or `node_modules` inside your  `C:\Users\tomsh`?

3. As per the second part - regarding `nativescript-angular` - have you tried running the sample inside `ng-sample` dir?

4. Can you reproduce the issue with a default template - `tns create myApp` ?

I'm looking forward to hearing from you.

**Comment by fant0m** [comment]
Hello,
1. Same [behavior](https://paste.ofcode.org/j5rrFjhqGywftvWNKVJpUh), I also included custom debug output but problem is related to 2.
2. I have only node_modules and that's causing an error, it's installing into `C:\Users\tomsh\node_modules `but loading from project folder.
3. Yeah, I tried. I got pretty same error message after running `tns build/run android`.
4. Yup, it's not related to Angular (same issue with js template).

**Comment by guessWho8** [comment]
Hello,
1. Same [behavior](https://paste.ofcode.org/j5rrFjhqGywftvWNKVJpUh), I also included custom debug output but problem is related to 2.
2. I have only node_modules and that's causing an error, it's installing into `C:\Users\tomsh\node_modules `but loading from project folder.
3. Yeah, I tried. I got pretty same error message after running `tns build/run android`.
4. Yup, it's not related to Angular (same issue with js template).

**Comment by rosen-vladimirov** [comment]
@guessWho8 ,
Thanks for the quick answers. It's really strange that the application is again trying to install the template in `C:\Users\tomsh`:

```
spawn: npm.cmd "install" "tns-template-hello-world-ng@3.0.0" "--silent" "--save-exact" "--save"
C:\Users\tomsh
```

It looks like the current working directory, passed [here](https://github.com/NativeScript/nativescript-cli/blob/master/lib/node-package-manager.ts#L197) is not working, but I'm not sure why.
Can you add a `console.log` statement there and see the value of the `cwd` argument.

After that, can you try installing CLI's `@next` version and see if the result is the same? Just execute `npm install -g nativescript@next`.

Thanks in advance for your help!

**Comment by guessWho8** [comment]
I'm sorry but node-package-manager.js has just 137 lines. Anyway I upgraded nativescript and now I'm getting this error: `cp: no such file or directory: D:\tns\test\node_modules\tns-template-hello-world-ng\*` since it's still installing to the root folder.

After upgrade I was able to insert console.log into that file. Here's result:

```
D:\tns>tns create test --ng
Installing  tns-template-hello-world-ng
**D:\tns\test // -> console.log(cwd)**
C:\Users\tomsh // -> this is not console.log(cwd)
`-- tns-template-hello-world-ng@3.1.0

cp: no such file or directory: D:\tns\test\node_modules\tns-template-hello-world-ng\*
```

**Comment by rosen-vladimirov** [comment]
Hey @guessWho8 ,
Can you try executing the follwing script somewhere in your `D:` drive
```
$ mkdir test1
$ cd test1
$ npm init -f
$ npm install lodash --save --save-exact
```
Can you confirm the `lodash` package is installed inside `...\test1\node_modules\lodash`?

I've tried to reproduce the issue, but still without success. I suppose there's some npm configuration on your side that makes it work in this way, but the above script should help us to understand if the problem is in CLI or in npm.

**Comment by guessWho8** [comment]
Hello,
yes, lodash package is installed inside test1. Actually, I have never had a problem with installing any npm package.

**Comment by guessWho8** [comment]
Hey @rosen-vladimirov, any news?

**Comment by rosen-vladimirov** [comment]
Hey @guessWho8 ,
Please excuse me for the late reply. Can you try executing the [following script](https://gist.github.com/rosen-vladimirov/40c258de1a366555ac36ceab66377f4d) - it should create two directories in the dir where you are executing it and should install lodash as dependency in both of them. Can you send me the output of the script?

Thanks again for your cooperation.

**Comment by guessWho8** [comment]
Hello @rosen-vladimirov,
here's the result:

```
D:\test2>node testNpm.js
Creating directory testDir.
Created directory testDir.
Creating package.json inside testDir.
Created package.json with full path D:\test2\testDir.
Installing lodash as dependency in D:\test2\testDir
#STDERR npm
#STDERR  WARN saveError ENOENT: no such file or directory, open 'C:\Users\tomsh\package.json'

#STDERR npm WARN
#STDERR enoent ENOENT: no such file or directory, open 'C:\Users\tomsh\package.json'
npm WARN tomsh No description
npm WARN tomsh No repository field.
npm WARN tomsh No README data
npm WARN tomsh No license field.


#STDOUT + lodash@4.17.4
added 1 package in 7.878s

#CLOSE 0
Installed lodash as dependency in D:\test2\testDir
(node:11556) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: ENOENT: no such file or directory, scandir 'D:\test2\testDir\node_modules'

```

**Comment by rosen-vladimirov** [comment]
@guessWho8 , thanks for the output - this confirms the problem is with the spawning `npm` commmands from Node.js.  Can you try upgrading your `npm` version by executing:
```
npm install -g npm@4
```
And try the script again?

**Comment by fant0m** [comment]
Unfortunately, the result is same. It's really weird.

**Comment by rosen-vladimirov** [comment]
Okay, can you try the following:
1. Uninstall npm globally: `npm uninstall -g npm`
2. Uninstall Node.js
3. Install latest Node.js 6.x.x version

And try the script again?

**Comment by guessWho8** [comment]
Same result. Probably we can close this issue since it's not related to nativescript. I will try to google some solution.

**Comment by rosen-vladimirov** [comment]
Okay @guessWho8 ,
You can try checking the `npm.cmd` file - maybe it has been modified and always tries to install something in incorrect place. 
Also make sure you are testing `npm install ...` manual executions in `cmd` not in `git bash` or some other terminals, as the Node.js is spawning `cmd` on Windows.

I'm closing this one, but please write down the solution once you find it.

**Comment by guessWho8** [comment]
Hello again @rosen-vladimirov,
finally, I was able to solve this issue! I thought it's not problem of npm itself, so I started playing with node's child_process. It was really weird because using your code and this:
`promisifiedChildProcess("pwd", [], { cwd: 'C://users//tomsh//test'})`
was returning correct path. I tried many options but later I also replaced `spawn` with `exec`:
`npmInstallChildProcess = childProcess.exec('pwd', {cwd: 'D:'});`
and I found out it's not actually changing the directory (it printed out C:/users/tomsh). So I tried google similar problems and I found this [one](https://github.com/nodejs/node/issues/6807#issuecomment-219992306). I also had that Autorun registry key with value C:/users/tomsh. I deleted it and now everything works correct. So I guess it's node's bug after all.
